contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -2540,6 +2540,119 @@ CREATE OR REPLACE FUNCTION sys.json_query(json_string text, path text default '$
 RETURNS sys.NVARCHAR
 AS 'babelfishpg_tsql', 'tsql_json_query' LANGUAGE C IMMUTABLE PARALLEL SAFE;
 
+--JSON_MODIFY
+CREATE OR REPLACE FUNCTION sys.json_modify(in expression NVARCHAR,in path_json TEXT, in new_value TEXT)
+RETURNS sys.NVARCHAR
+AS
+$BODY$
+DECLARE
+    json_path TEXT;
+    json_path_convert TEXT;
+    new_jsonb_path TEXT[];
+    type_key_value TEXT;
+    mode TEXT;
+    modifier TEXT;
+    len_array INTEGER;
+    word_count INTEGER;
+    create_if_missing BOOL:='TRUE';
+    append_modifier BOOL:='FALSE';
+    key_exist BOOL;
+    key_value JSONB;
+    json_expression JSONB;
+    result_json NVARCHAR;
+BEGIN
+    json_expression = to_jsonb(expression::JSONB);
+
+    json_path = regexp_replace(path_json, '^.* ', '','ig'); -- To select the last word of the json_path string. \\\\\\Instead of this one can use split_part if regexp is slow
+    json_path_convert = regexp_replace(json_path, '\$\.|]|\$\[' , '' , 'ig'); -- To remove the "$." and "]" sign from the string 
+    json_path_convert = regexp_replace(json_path_convert, '\.|\[' , ',' , 'ig'); -- To replace the "." and "[" with the "," to change into required format
+    new_jsonb_path = CONCAT('{',json_path_convert,'}'); -- Final required format of path by jsonb_set
+    
+    
+    key_exist = jsonb_path_exists(json_expression,json_path::jsonpath); -- TO check if key exist in the given path
```

Comment:
For append modifier:
If the key exists does not exists then we need to convert the new_value as array and then add it in the expression or else we do not need to convert the new_value to array_type and also we only check the value of the key only if the key_exists.

If append modifier is not present:
for strict mode the function works in different way if the key_exists or not. It returns an error if key exists and returns a new value in case key_exists.

Overall key_exists makes it easy to divide working of the function in different cases.